### PR TITLE
Smooth initial crash volume

### PR DESCRIPTION
**Pull request by howard-stearns**
For initial contact in a crash, use the change in object velocities rather than the penetration.
This gives more consistent results when the contact happens at different parts of the simulation cycle.

Continuing contact is left unchanged.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: c036b5fd4ba759a574e24d0463d3f75b16518497 available at: https://gist.github.com/2c3288b97a0c5f9c0dd1<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5033.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5033.dmg

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -1131,10 +1131,15 @@ void EntityTreeRenderer::playEntityCollisionSound(const QUuid& myNodeID, EntityT
     }
     const float mass = entity->computeMass();
     const float COLLISION_PENTRATION_TO_VELOCITY = 50; // as a subsitute for RELATIVE entity->getVelocity()
-    const float linearVelocity = glm::length(collision.penetration) * COLLISION_PENTRATION_TO_VELOCITY;
+    // The collision.penetration is a pretty good indicator of changed velocity AFTER the initial contact,
+    // but that first contact depends on exactly where we hit in the physics step.
+    // We can get a more consistent initial-contact energy reading by using the changed velocity.
+    const float linearVelocity = (collision.type == CONTACT_EVENT_TYPE_START) ?
+        glm::length(collision.velocityChange) :
+        glm::length(collision.penetration) * COLLISION_PENTRATION_TO_VELOCITY;
```
Comment (PhilipRosedale) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: spelling typo... PENTRATION

Comment (howard-stearns) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: We should avoid problems in magic number constants by having constants for words which are then used to create symbol names.

Just kidding. Will fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 407746b842b7c3d2aaf01d7cc5ee23962425d58c available at: https://gist.github.com/2c3288b97a0c5f9c0dd1<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5033.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5033.dmg

File: libraries/physics/src/ObjectMotionState.cpp [hunk 2]
```diff
@@ -82,6 +82,9 @@ void ObjectMotionState::setBodyGravity(const glm::vec3& gravity) const {
 glm::vec3 ObjectMotionState::getBodyLinearVelocity() const {
     return bulletToGLM(_body->getLinearVelocity());
 }
+glm::vec3 ObjectMotionState::getObjectLinearVelocityChange() const {
+    return glm::vec3();  // Subclasses override where meaningful.
+}
```
Comment (AndrewMeadows) [inline libraries/physics/src/ObjectMotionState.cpp]: I don't think the default glm::vec3 ctor doesn't actually initialize the component data.  Best to use the ctor that takes one float argument:

    return glm::vec3(0.0f);

**Comment by hifi-gustavo** [comment]
Build console output for commit: 407d3b6a6eff1461d54dd6ff499b3d5e40bd94a5 available at: https://gist.github.com/2c3288b97a0c5f9c0dd1<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5033.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5033.dmg

**Comment by AndrewMeadows** [comment]
I'm going to merge this so we can try to demo it today, but now that I look it over a second time I have some minor ideas on how to make it better.  I'll make an asana task for those.

**Pull request merged**